src/org/rascalmpl/util/maven/SimpleResolver.java

```diff
@@ -51,10 +54,17 @@
     private final ModelBuilder builder;
     private final HttpClient client;
 
+    private Map<String, Mirror> mirrors;
+
     public SimpleResolver(Path rootRepository, ModelBuilder builder, HttpClient client) {
         this.rootRepository = rootRepository;
         this.builder = builder;
         this.client = client;
+
+        mirrors = new HashMap<>();
+        for (Mirror mirror : Util.getMirrors()) {
```

Comment:
this now means we have to reparse the settings.xml every time we build a new resolver (which we rebuild for every maven parse).

I think this should be moved upwards. just like the `rootRepository` which is also defined from the `settings.xml`

Reply:
I have moved the reading of settings up. This also improves some of the tests as they are no longer dependent on the content of the settings.xml of the user: the settings are injected into `MavenParser` through the constructor.